File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/OrderController.java
package com.pinyougou.manager.controller;


import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.order.service.OrderService;
import com.pinyougou.pojo.TbOrder;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/order")
public class OrderController {

    @Reference
    private OrderService orderService;




    /**
     * 查询所有订单
     * @return list<TbOrder>
     */
    @RequestMapping("/findAll")
    public List<TbOrder> findAll(){

        return orderService.findAll();
    }


    /**
     * 返回全部列表
     * @return
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page, int rows){
        return orderService.findPage(page, rows);
    }

    /**
     * 查询+分页
     * @param order
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody TbOrder order, int page, int rows  ){
        return orderService.findPage(order, page, rows);
    }


    /**
     * 根据id返回一列信息
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public TbOrder findOne(Long id){
        System.out.println(id);
        TbOrder tbOrder = orderService.findOne(id);
        System.out.println(tbOrder.toString());
        return tbOrder;
    }

}

File: pinyougou-manager-web/src/main/webapp/js/controller/orderController.js
 //控制层 
app.controller('orderController' ,function($scope,$controller,orderService){
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
        orderService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){
        orderService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){
        orderService.findOne(id).success(
			function(response){//tbOrder
				$scope.tbOrder= response;
			}
		);				
	}

	


    $scope.searchEntity={auditStatus:'0'};//定义搜索对象
	
	//搜索
	$scope.search=function(page,rows){
        orderService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
    $scope.status=['未付款','已付款','未发货','已发货','交易成功','交易关闭','已发货'];
    $scope.type=['在线支付','货到付款']
    
});	
